src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -619,7 +619,8 @@ void SpatialPooler::compute(UInt inputArray[], bool learn,
   if (learn) {
     boostOverlaps_(overlaps_, boostedOverlaps_);
   } else {
-    boostedOverlaps_.assign(overlaps_.begin(), overlaps_.end());
+    for (auto overlap : overlaps_)
+      boostedOverlaps_.push_back(overlap);
```

Comment:
Good call on clearing array. This was relating to float vs double with ```NTA_DOUBLE_PRECISION``` being always set in Windows version. That flag was removed a while back during this PR, will revert this change back to the assign.